src/yesno.ts

```diff
@@ -278,7 +278,11 @@ export class YesNo implements IFiltered {
   private async onIntercept(event: IInterceptEvent): Promise<void> {
     this.recordRequest(event.requestSerializer, event.requestNumber);
 
-    if (!this.ctx.hasResponsesDefinedForMatchers() && !this.isMode(Mode.Mock)) {
+    if (
+      (!this.ctx.hasResponsesDefinedForMatchers() && !this.isMode(Mode.Mock)) ||
+      (this.ctx.hasIgnoresDefinedForMatchers() &&
```

Comment:
Could this be simplified by just checking `hasMatchingIgnore` instead of also checking `hasIgnoresDefinedForMatches`?

You could just update `hasMatchingIgnore` immediatly return if `ignoresForMatchingRequests` is empty.

Reply:
I'd also break this into two if statements. It's not as concise but it's easier to read in my opinion and you can put a comment as to what is going on in each.

```
// proxy request when ...
if 

// proxy request if explicitly ignored
if

Reply:
I simplified the hasMatchingIgnores logic and split up the if statements for clarity.